wowchemy/layouts/slides/baseof.html

```diff
@@ -20,55 +21,176 @@
 
-  {{ $css := resources.Get "css/reveal.css" }}
   {{ $css_custom := resources.Get "css/reveal_custom.css" }}
-  {{ $style := slice $css $css_custom | resources.Concat "css/reveal_custom.css" | resources.Minify }}
+  {{ $style := slice $css_custom | resources.Concat "css/reveal_custom.css" | resources.Minify }}
   <link rel="stylesheet" href="{{ $style.RelPermalink }}">
 
-  <!-- Printing and PDF exports -->
-  <script>
-    var link = document.createElement( 'link' );
-    link.rel = 'stylesheet';
-    link.type = 'text/css';
-    link.href = window.location.search.match( /print-pdf/gi ) ? '{{ $cdn_url_reveal }}/css/print/pdf.css' : '{{ $cdn_url_reveal }}/css/print/paper.css';
-    document.getElementsByTagName( 'head' )[0].appendChild( link );
-  </script>
-
 </head>
 <body>
 
   {{ block "main" . }}{{ end }}
 
-  <script src="{{ $cdn_url_reveal }}/js/reveal.min.js"></script>
+  <script src="{{ $cdn_url_reveal }}/reveal.min.js"></script>
+
+  <script src="{{ $cdn_url_reveal }}/plugin/markdown/markdown.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/highlight/highlight.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/notes/notes.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/search/search.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/math/math.js"></script>
+  <script src="{{ $cdn_url_reveal }}/plugin/zoom/zoom.js"></script>
+
+<!-- 3rd part plugins -->
+
+  <script src="/js/vendor/reveal.js/plugin/menu/menu.js"></script>
+  <script src="/js/vendor/reveal.js/plugin/chalkboard/plugin.js"></script>
+  {{- $diagram := $.Param "slides.diagram" | default false -}}
+  {{ if $diagram }}
+    {{ printf "<script src=\"%s\" integrity=\"%s\" crossorigin=\"anonymous\" title=\"mermaid\"></script>" (printf $js.mermaid.url $js.mermaid.version) $js.mermaid.sri | safeHTML }}
+  {{ end }}
+
+  <!-- reveal configuration -->
+  <!-- Set configuration template variables and their defaults -->
+  {{- $controls := $.Param "slides.controls" | default true -}}
+  {{- $progress := $.Param "slides.progress" | default true -}}
+  {{- $slideNumber := $.Param "slides.slideNumber" | default false -}}
+  {{- $center := $.Param "slides.center" | default true -}}
+  {{- $rtl := $.Param "slides.rtl" | default false -}}
+  {{- $mouseWheel := $.Param "slides.mouseWheel" | default true -}}
+  {{- $transition := $.Param "slides.transition" | default "'slide'" -}}
+  {{- $transitionSpeed := $.Param "slides.transitionSpeed" | default "'default'" -}}
+  {{- $backgroundTransition := $.Param "slides.backgroundTransition" | default "'fade'" -}}
+  {{- $menu := $.Param "slides.menu" | default true -}}
+  {{- $chalkboard := $.Param "slides.chalkboard" | default true -}}
+  {{- $touch := $.Param "slides.touch" | default true -}}
+  {{- $loop := $.Param "slides.loop" | default true -}}
 
+<!-- The following is just to set the proper configurations based on slides front matter to initialize reveal -->
   <script>
```

Comment:
Please refactor the JS to a dedicated file and use Hugo's new JS Param system rather than using Go templating without JS code.